pkg/karpenter/karpenter.go

```diff
@@ -0,0 +1,69 @@
+package karpenter
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/kris-nova/logger"
+
+	"github.com/weaveworks/eksctl/pkg/karpenter/providers"
+)
+
+const (
+	// default namespace for Karpenter
+	KarpenterNamespace = "karpenter"
+
+	karpenterHelmRepo         = "https://charts.karpenter.sh"
+	karpenterHelmChartName    = "karpenter/karpenter"
+	karpenterReleaseName      = "karpenter"
+	controllerClusterName     = "controller.clusterName"
+	controllerClusterEndpoint = "controller.clusterEndpoint"
+	createServiceAccount      = "serviceAccount.create"
+	addDefaultProvisioner     = "defaultProvisioner.create"
+)
+
+// Options contains values which Karpenter uses to configure the installation.
+type Options struct {
+	HelmInstaller         providers.HelmInstaller
+	Namespace             string
+	ClusterName           string
+	AddDefaultProvisioner bool
+	CreateServiceAccount  bool
+	ClusterEndpoint       string
+	Version               string
+}
+
+// Handler defines a handler for Karpenter.
+type Handler interface {
```

Comment:
:vomiting_face: handler sounds like an API handler :joy:

Reply:
@aclevername I've been thinking what to name this f8cker for the past HOUR! :D PLEASE!! HALP! :D

Reply:
@aclevername `Controller` `Overseer` `Builder`

Reply:
@aclevername I went with an alternative approach. Would you mind checking if you like that? :)